RISTNet.cpp

```diff
@@ -153,12 +159,18 @@ int RISTNetReceiver::clientDisconnect(void *pArg, rist_peer *pPeer) {
         return 0;
     }
 
-    if (lWeakSelf->mClientList.find(pPeer) == lWeakSelf->mClientList.end()) {
+	auto netObj = lWeakSelf->mClientList.find(pPeer);
+	if (netObj == lWeakSelf->mClientList.end()) {
         LOGGER(true, LOGG_ERROR, "RISTNetReceiver::clientDisconnect unknown peer")
         return 0;
-    } else {
-        lWeakSelf->mClientList.erase(lWeakSelf->mClientList.find(pPeer)->first);
+	}
+
+    if (lWeakSelf->clientDisconnectedCallback) {
+        lWeakSelf->clientDisconnectedCallback(netObj->second, *pPeer);
+		lWeakSelf->clientDisconnectedCallback = nullptr;
```

Comment:
Is it possible to reconnect in RIST? In that case you might want to not reset the disconnect callback.

Reply:
Good catch! I'll check.

Reply:
This is not correct. The receiver can have multiple connections. If one disconnects, the callback should not be reset. And even if the receiver just have one sender, another can connect later.

Reply:
:+1: